test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -63,10 +63,14 @@ build_branch()
     echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
     echo "Building branch: \"$1\""
     echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+    
+    # Only clone the branch if the option is picked
+    if [ "$2" == 1 ]; then
```

Comment:
I'm trying to understand the use case of the newly introduced 2nd arg in this function. Is it to save the repo clone when running the script manually multiple times? If that's the case, please note the pre-cloned local repo could contain stale data the next time you run the script (e.g. someone merges a commit into the branch you are testing against). I think it's safer to either clone it freshly every time to get the latest changes, or handle the potential stale data problem within the function (e.g. git pull after checkout the branch), instead of introducing a new arg. And btw, the new arg value 1 does not feel quite informational.

Reply:
The addition of the 2nd arg here is to allow for the recompilation of the code without cloning it from git. The reason is that the patch is applied to the cloned branch and the test is then run on that so if we were to use the build_branch code as it is, the added patch would be re-written by the cloning of the branch. This is why I decided to separate cloning and building, and passed an arg value as 1 to signify clone = 1 and not cloning = 0. I can see how that could've been more informational and less "magic-number" though. Let me know your thoughts!

Reply:
After reviewing the changes down there, I think I understand the use case being the need for recompiling after applying the patch (line 705). I do prefer containing the handling within the function over introducing a new arg :)

Reply:
Would splitting the `build_branch` function into two ( `clone_branch` and `build_branch`) be feasible? Then wherever `build_branch` is currently used call both of the functions.

Reply:
That could be a good solution as well 👍